### PR TITLE
python310Packages.altair: 4.2.2 -> 5.0.1

### PR DESCRIPTION
**Pull request by Vinetos**
###### Description of changes
This MR update the altair Python package to the new major version.

This MR changed to way to build the package from Pypi to Github because the Pypi package does not more include tests (200+ tests).

Full changelog here: https://github.com/altair-viz/altair/compare/v4.2.2...v5.0.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/altair/default.nix [hunk 1]
```diff
@@ -1,62 +1,77 @@
...
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-OTmaJnxJsw0QLBBBHmerJjdBVqhLGuufzRUUBCm6ScU=";
+  # We use fetch from Github because the Pypi is missing the tests
+  src = fetchFromGitHub {
+    owner = "altair-viz";
+    repo = "${pname}";
```
Comment (natsukium) [inline pkgs/development/python-modules/altair/default.nix:36]:
```suggestion
    repo = "altair";
```

File: pkgs/development/python-modules/altair/default.nix [hunk 2]
```diff
@@ -1,62 +1,77 @@
...
+    rev = "ed1d42dd5d9c689803e172d36a39d6db76aadeb9";
```
Comment (natsukium) [inline pkgs/development/python-modules/altair/default.nix:37]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/development/python-modules/altair/default.nix [hunk 3]
```diff
@@ -1,62 +1,77 @@
...
+    hash = "sha256-7bTrfryu4oaodVGNFNlVk9vXmDA5/9ahvCmvUGzZ5OQ=";
   };
 
   propagatedBuildInputs = [
-    entrypoints
+    hatchling
```
Comment (natsukium) [inline pkgs/development/python-modules/altair/default.nix:42]: `hatchling` is only needed at build time.
It should go to `nativeBuildInputs`.

File: pkgs/development/python-modules/altair/default.nix [hunk 4]
```diff
@@ -1,62 +1,77 @@
...
-  ] ++ lib.optionals (pythonOlder "3.5") [ typing ];
+  ] ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata ]
+    ++ lib.optionals (pythonOlder "3.11") [ typing-extensions ];
 
   nativeCheckInputs = [
-    glibcLocales
+    hatch
+    ruff
+    black
+    mypy
```
Comment (natsukium) [inline pkgs/development/python-modules/altair/default.nix:55]: It's hard to see the test dependencies, but it looks like these are unnecessary.

File: pkgs/development/python-modules/altair/default.nix [hunk 5]
```diff
@@ -1,62 +1,77 @@
...
   meta = with lib; {
     description = "A declarative statistical visualization library for Python.";
-    homepage = "https://github.com/altair-viz/altair";
+    homepage = "https://altair-viz.github.io";
     license = licenses.bsd3;
-    maintainers = with maintainers; [ teh ];
+    maintainers = with maintainers; [ teh vinetos ];
     platforms = platforms.unix;
   };
+
```
Comment (natsukium) [inline pkgs/development/python-modules/altair/default.nix:76]:
```suggestion
```

File: pkgs/development/python-modules/altair/default.nix [hunk 6]
```diff
@@ -1,62 +1,77 @@
...
+      "tests/vegalite/v5/test_api.py" # TODO: Disabled because of missing altair_viewer package
+  ];
 
   meta = with lib; {
     description = "A declarative statistical visualization library for Python.";
-    homepage = "https://github.com/altair-viz/altair";
+    homepage = "https://altair-viz.github.io";
     license = licenses.bsd3;
-    maintainers = with maintainers; [ teh ];
+    maintainers = with maintainers; [ teh vinetos ];
```
Comment (natsukium) [inline pkgs/development/python-modules/altair/default.nix:73]: Adding the maintainer and the version bump should be separate commits.

Then please don't forget to add yourself to the maintainer list.

File: pkgs/development/python-modules/altair/default.nix (hunk 4 above)
Comment (Vinetos) [reply pkgs/development/python-modules/altair/default.nix:55]: You are right, Only `black` is used in `tests/test_toplevel.py`

**Comment by natsukium** [comment]
Would you please rebase it again to fix the ofborg error?

**Comment by Vinetos** [comment]
Done :smile:

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 237915` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>streamlit</li>
    <li>streamlit.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.altair</li>
    <li>python310Packages.altair.dist</li>
    <li>python311Packages.altair</li>
    <li>python311Packages.altair.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
streamlit supports altair5 at https://github.com/streamlit/streamlit/pull/6618.
This update looks fine.

**Comment by Vinetos** [comment]
Any update on when it could be merged ?

**Comment by Janik-Haag** [comment]
```
./results/streamlit/bin/streamlit 
Traceback (most recent call last):
  File "/nix/store/wcgsxyflkp22zllhpprsc0m2zmxyb1ph-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/core/__init__.py", line 23, in <module>
    from . import multiarray
  File "/nix/store/wcgsxyflkp22zllhpprsc0m2zmxyb1ph-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/core/multiarray.py", line 10, in <module>
    from . import overrides
  File "/nix/store/wcgsxyflkp22zllhpprsc0m2zmxyb1ph-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/core/overrides.py", line 6, in <module>
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/bin/.streamlit-wrapped", line 6, in <module>
    from streamlit.web.cli import main
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/__init__.py", line 55, in <module>
    from streamlit.delta_generator import DeltaGenerator as _DeltaGenerator
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/delta_generator.py", line 36, in <module>
    from streamlit import config, cursor, env_util, logger, runtime, type_util, util
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/cursor.py", line 18, in <module>
    from streamlit.runtime.scriptrunner import get_script_run_ctx
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/runtime/__init__.py", line 16, in <module>
    from streamlit.runtime.runtime import Runtime as Runtime
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/runtime/runtime.py", line 31, in <module>
    from streamlit.runtime.app_session import AppSession
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/runtime/app_session.py", line 35, in <module>
    from streamlit.runtime import caching, legacy_caching
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/runtime/caching/__init__.py", line 21, in <module>
    from streamlit.runtime.caching.cache_data_api import (
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/runtime/caching/cache_data_api.py", line 32, in <module>
    from streamlit.runtime.caching.cache_errors import CacheError, CacheKeyNotFoundError
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/runtime/caching/cache_errors.py", line 18, in <module>
    from streamlit import type_util
  File "/nix/store/b0jjxhfavfpiapjasmw8sx9mdj2507hf-streamlit-1.24.0/lib/python3.10/site-packages/streamlit/type_util.py", line 40, in <module>
    import numpy as np
  File "/nix/store/wcgsxyflkp22zllhpprsc0m2zmxyb1ph-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/__init__.py", line 141, in <module>
    from . import core
  File "/nix/store/wcgsxyflkp22zllhpprsc0m2zmxyb1ph-python3.11-numpy-1.24.2/lib/python3.11/site-packages/numpy/core/__init__.py", line 49, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.10 from "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/bin/python3.10"
  * The NumPy version is: "1.24.2"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'
```
streamlit has some errors with numpy after updating, they seem to be related becuase they don't happen on master or when reseting the commit.

**Review by Janik-Haag — COMMENTED** [review]
changes look good except for the wired stream lit failure.

**Comment by natsukium** [comment]
I have seen the same problem before.
https://github.com/NixOS/nixpkgs/pull/231095#issuecomment-1542921767

but I'm not sure what the reason is.

**Review by Janik-Haag — COMMENTED** [review]
please change the commit adding you as a maintainer to `maintainers: add vinetos` otherwise this looks good.

**Comment by Janik-Haag** [comment]
> I have seen the same problem before. [#231095 (comment)](https://github.com/NixOS/nixpkgs/pull/231095#issuecomment-1542921767)
> 
> but I'm not sure what the reason is.

I have seen the error related to nixpkgs-review but it also occurred here for me when just using nix-build :confused: 
I don't really have an idea on how we could fix that.

**Comment by Vinetos** [comment]
> please change the commit adding you as a maintainer to `maintainers: add vinetos` otherwise this looks good.

Done

File: pkgs/development/python-modules/altair/default.nix [hunk 7]
```diff
@@ -1,62 +1,70 @@
...
 
   pythonImportsCheck = [ "altair" ];
 
-  # avoid examples directory, which fetches web resources
-  preCheck = ''
-    cd altair/tests
-  '';
+  disabledTestPaths = [
+      "tests/test_examples.py" # Disabled because it uses network
+      "tests/vegalite/v5/test_api.py" # TODO: Disabled because of missing altair_viewer package
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altair/default.nix:60]:
```suggestion
      # Disabled because it requires internet connectivity
      "tests/test_examples.py" 
      # TODO: Disabled because of missing altair_viewer package
      "tests/vegalite/v5/test_api.py"
```

File: pkgs/development/python-modules/altair/default.nix [hunk 8]
```diff
@@ -1,62 +1,70 @@
...
     toolz
-    jinja2
-  ] ++ lib.optionals (pythonOlder "3.5") [ typing ];
+  ] ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata ]
+    ++ lib.optionals (pythonOlder "3.11") [ typing-extensions ];
 
   nativeCheckInputs = [
-    glibcLocales
+    hatchling
+    black
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altair/default.nix:49]:
```suggestion
```
that's a linter which we don't want to fail on

File: pkgs/development/python-modules/altair/default.nix [hunk 9]
```diff
@@ -1,62 +1,70 @@
...
-    six
     toolz
-    jinja2
-  ] ++ lib.optionals (pythonOlder "3.5") [ typing ];
+  ] ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata ]
+    ++ lib.optionals (pythonOlder "3.11") [ typing-extensions ];
 
   nativeCheckInputs = [
-    glibcLocales
+    hatchling
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altair/default.nix:48]: hatchling is used to build the package, so it should go to nativeBuildInputs

File: pkgs/development/python-modules/altair/default.nix [hunk 10]
```diff
@@ -1,62 +1,70 @@
...
+    jinja2
     jsonschema
     numpy
     pandas
-    six
     toolz
-    jinja2
-  ] ++ lib.optionals (pythonOlder "3.5") [ typing ];
+  ] ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata ]
+    ++ lib.optionals (pythonOlder "3.11") [ typing-extensions ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altair/default.nix:45]:
```suggestion
  ] ++ lib.optional (pythonOlder "3.8") importlib-metadata
    ++ lib.optional (pythonOlder "3.11") typing-extensions;
```
nit

File: pkgs/development/python-modules/altair/default.nix [hunk 11]
```diff
@@ -1,62 +1,70 @@
...
-  version = "4.2.2";
-  disabled = isPy27;
+  version = "5.0.1";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-OTmaJnxJsw0QLBBBHmerJjdBVqhLGuufzRUUBCm6ScU=";
+  # We use fetch from Github because the Pypi is missing the tests
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altair/default.nix:30]:
```suggestion
```
not commenty worthy

File: pkgs/development/python-modules/altair/default.nix [hunk 12]
```diff
@@ -1,62 +1,70 @@
...
-  '';
+  disabledTestPaths = [
+      "tests/test_examples.py" # Disabled because it uses network
+      "tests/vegalite/v5/test_api.py" # TODO: Disabled because of missing altair_viewer package
+  ];
 
   meta = with lib; {
     description = "A declarative statistical visualization library for Python.";
-    homepage = "https://github.com/altair-viz/altair";
+    homepage = "https://altair-viz.github.io";
```
[the remark this answers is not in the document]
Comment (Vinetos) [reply pkgs/development/python-modules/altair/default.nix:65]:
```suggestion
    homepage = "https://altair-viz.github.io";
    downloadPage = "https://github.com/altair-viz/altair";
```

**Comment by Vinetos** [comment]
Suggestions and modifications applied, Ready to be reviewed

File: pkgs/development/python-modules/altair/default.nix [hunk 13]
```diff
@@ -1,62 +1,74 @@
...
 
-  # avoid examples directory, which fetches web resources
-  preCheck = ''
-    cd altair/tests
-  '';
+  disabledTestPaths = [
+      # Disabled because it requires internet connectivity
+      "tests/test_examples.py"
+      # TODO: Disabled because of missing altair_viewer package
+      "tests/vegalite/v5/test_api.py"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altair/default.nix:63]:
```suggestion
    # Disabled because it requires internet connectivity
    "tests/test_examples.py"
    # TODO: Disabled because of missing altair_viewer package
    "tests/vegalite/v5/test_api.py"
```
according to our contributing guide we use 2 spaces for nix files

File: pkgs/development/python-modules/altair/default.nix [hunk 14]
```diff
@@ -1,62 +1,74 @@
...
 
   meta = with lib; {
     description = "A declarative statistical visualization library for Python.";
-    homepage = "https://github.com/altair-viz/altair";
+    homepage = "https://altair-viz.github.io";
+    downloadPage = "https://github.com/altair-viz/altair";
     license = licenses.bsd3;
-    maintainers = with maintainers; [ teh ];
+    maintainers = with maintainers; [ teh vinetos ];
     platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altair/default.nix:72]:
```suggestion
```
I only just noticed: that is the default for buildPythonPackage and should not be repeated

File: pkgs/development/python-modules/altair/default.nix [hunk 15]
```diff
@@ -1,62 +1,74 @@
...
     numpy
     pandas
-    six
     toolz
-    jinja2
-  ] ++ lib.optionals (pythonOlder "3.5") [ typing ];
+  ] ++ lib.optional (pythonOlder "3.8") importlib-metadata
+    ++ lib.optional (pythonOlder "3.11") typing-extensions;
+
+  nativeBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altair/default.nix:46]: normally I would swap the positions of nativeBuildInputs and propagatedBuildInputs here but more a nit

**Review by SuperSandro2000 — COMMENTED** [review]
didn't really want to do another round but found two minor things and I can't squash merge because of the maintainer addition which should be kept extra

**Comment by Vinetos** [comment]
> didn't really want to do another round but found two minor things and I can't squash merge because of the maintainer addition which should be kept extra

No problem. MR should be fine :D

File: pkgs/development/python-modules/altair/default.nix [hunk 16]
```diff
@@ -1,62 +1,73 @@
...
 
-  # avoid examples directory, which fetches web resources
-  preCheck = ''
-    cd altair/tests
-  '';
+  disabledTestPaths = [
+    # Disabled because it requires internet connectivity
+    "tests/test_examples.py"
+    # TODO: Disabled because of missing altair_viewer package
+    "tests/vegalite/v5/test_api.py"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altair/default.nix:63]:
```suggestion
    "tests/vegalite/v5/test_api.py"
    # avoid updating files and dependency on black
    "tests/test_toplevel.py"
```
we need to disable a test file to not require black

Comment (Vinetos) [reply pkgs/development/python-modules/altair/default.nix:63]: Should I remove black from input too ?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/altair/default.nix:63]: yes

Comment (Vinetos) [reply pkgs/development/python-modules/altair/default.nix:62]: done

**Comment by happysalada** [comment]
I'm getting consistent failures of
```
FAILED tests/vegalite/v5/test_params.py::test_selection_deprecation - Deprecati
onWarning: jsonschema.RefResolver is deprecated as of v4.18.0, in
```
on linux-x86_64

just saying in case someone is interested.
